src/pages/users/index.tsx

```diff
@@ -0,0 +1,71 @@
+import {FunctionComponent, useEffect, useState} from 'react'
+import {useQuery} from 'react-query'
+
+import {UserCard} from '@/components/pages/users/cards/users'
+import {ArrowForward} from '@/components/icons/arrows'
+import {fetchGroups, fetchRoles, fetchUsers} from '@/pages/api/users'
+
+import type {IGroup, IRole, IUser} from '@/types'
+
+const Users: FunctionComponent = () => {
+  const [totalUsers, setTotalUsers] = useState(0)
+  const [totalGroups, setTotalGroups] = useState(0)
+  const [totalRoles, setTotalRoles] = useState(0)
+
+  const {isLoading, data: usersData} = useQuery<Record<string, IUser>, Error>('users', fetchUsers)
+
+  const {data: groupsData} = useQuery<Record<string, IGroup>, Error>('groups', fetchGroups)
+  const {data: rolesData} = useQuery<Record<string, IRole>, Error>('roles', fetchRoles)
+
+  useEffect(() => {
```

Comment:
I failed to see why we need to keep states to track the total. Is it possible to replace this effect with a direct call to each of the deps?